core/src/main/java/org/jruby/RubyClass.java

```diff
@@ -307,6 +309,49 @@ public int getVariableTableSizeWithExtras() {
         return new HashMap<String, VariableAccessor>(getVariableAccessorsForRead());
     }
 
+    /**
+     * Request that the listed ivars (no @ in name) have field storage when we are reified
+     * @param names
+     */
+    public synchronized void requestFieldStorage(String name, Class<?> fieldType,  Boolean unwrap, Class<?> toType) {
+        AdvancedFieldConfiguration config = new JavaClassConfiguration.AdvancedFieldConfiguration(name, fieldType, unwrap, toType);
+        if (reifiedClass != null)
+            requestFieldStorage(config);
+        else {
+            if (javaClassConfiguration.requestedStorageVariables == null)
+                javaClassConfiguration.requestedStorageVariables = new ArrayList<>();
+            javaClassConfiguration.requestedStorageVariables.add(config);
+        }
+    }
+
+    /**
+     * Actually requests field storage for the ivar once we are reified
+     */
+    private void requestFieldStorage(AdvancedFieldConfiguration config) {
```

Comment:
RubyClass seems to be accumulating a lot of clutter related to reification that isn't strictly part of the RubyClass behavior. Almost feels like this stuff should live in the VariableTableManager rather than here. Might at least be worth making the method above private  (if possible) until we decide how to reorg this logic.

Reply:
I've moved this to VariableTableManager, though I had to make it public so RubyClass could access it. The other method needs to be public as it's configured in class.rb

Reply:
Ok... we can go with that and think about refactoring in the future. Nobody should call this stuff outside of JRuby anyway.